### DISCUSSION TITLE
Parameter import doesn't always work. (To discuss): UI feedback December 2022

### DISCUSSION THREAD

**Discussion by amc1999** [openmpp/UI#31] [General]
![unnamed](https://github.com/openmpp/UI/assets/29126882/bfdf17da-1d6f-480d-8825-9e0033baa813)

Please send us an example.

**Comment by rochellegarner** [comment]
Hi @amc1999 and @openmpp-org, just following up on this item, as we continue to have challenges with importing parameters. As an example, I tried to import an updated set of values for breast cancer treatments (customBreastCancerTreatmentCostNoScoSurg) into OncoSim v3.6.1.4 on my work tower. If the new parameter settings are in a file with a different file name, I get the following error messages: 
![image](https://github.com/openmpp/UI/assets/96252923/9cd93b19-8668-4c95-b4e9-33009b3a72c3)

If the filename is exactly the same, I get the following error message:
![image](https://github.com/openmpp/UI/assets/96252923/f7ca5352-00f4-45a8-9a80-19b8c7229a1b)

Perhaps I am doing something wrong? Or is there an error in the program?

**Comment by amc1999** [comment]
There are 3 options for updating parameter(s):

1. Simple for a small parameters: copy all or some parameter values from clipboard by using Ctrl + V
2. Upload ParameterName.csv file, file name has to be exactly the same as parameter name, for example: AgeBaselinePrge1.csv

![image](https://github.com/openmpp/UI/assets/29126882/3ba0bce4-01c1-484e-b4b2-7554a649af62)

3. Upload new input scenario .zip with parameter(s) CSV files.
  - it has to .zip file named as: ModelName.set.ScenarioName.zip, for example: RiskPaths.set.New.zip
  - inside of that .zip it can have:
    - scenario metadata .json file (optional)
    - folder with parameter(s) .csv file(s)
You can download Default scenario .zip to see a full example of scenario .zip

It is possible to create scenario wit only one single parameter .csv

For example we can create New scenario for RiskPaths model with SimulationCases.csv parameter
- create directory RiskPaths.set.New
- inside of above create directory set.New
- put SimulationCases.csv into RiskPaths.set.New/set.New directory
- make a zip: RiskPaths.set.New.zip
- upload it

![image](https://github.com/openmpp/UI/assets/29126882/0a67546e-76fa-47ed-b5e0-a359604bdb3e)

![image](https://github.com/openmpp/UI/assets/29126882/1307c3b7-e42f-4838-a717-265900b7c66c)

![image](https://github.com/openmpp/UI/assets/29126882/262cae9a-09e3-427f-ac4d-82a781ad9053)

![image](https://github.com/openmpp/UI/assets/29126882/052a6079-d4e4-4bc3-abb4-40805f5bf93c)




    

    -

**Comment by rochellegarner** [comment]
Ok, thanks Anatoly. From the wiki, it appeared to indicated that I could upload a single parameter file, including CSV (OpenM++ wiki (openmpp.org)<https://openmpp.org/wiki/openmpp-wiki.html#Ompp-UI-Upload-Parameter-or-Scenario>)

***@***.***

Are you saying that this isn't, in fact, functional?

Thanks

Rochelle

**Comment by amc1999** [comment]
Status update: It is look an Excel saving into CSV issue. 

Thank you, Rochelle for resolving that problem. 
It was not obvious, and it was a time consuming investigation.
